File: VsIntegration/ItemTemplates/SpecFlowStepDefinition_VB/SpecFlowStepDefinition1.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports TechTalk.SpecFlow

Namespace $rootnamespace$

    <Binding()> _
    Public Class $safeitemname$

        <Given("I have entered (.*) into the calculator")> _
        Public Sub GivenIHaveEnteredSomethingIntoTheCalculator(ByVal number As Integer)
            'TODO: implement arrange (recondition) logic
            ' For storing and retrieving scenario-specific data, 
            ' the instance fields of the class or the
            '     ScenarioContext.Current
            ' collection can be used.
            ' To use the multiline text or the table argument of the scenario,
            ' additional string/Table parameters can be defined on the step definition
            ' method. 

            ScenarioContext.Current.Pending()
        End Sub

        <[When]("I press add")> _
        Public Sub WhenIPressAdd()
            'TODO: implement act (action) logic

            ScenarioContext.Current.Pending()
        End Sub

        <[Then]("the result should be (.*) on the screen")> _
        Public Sub ThenTheResultShouldBe(ByVal result As Integer)
            'TODO: implement assert (verification) logic

            ScenarioContext.Current.Pending()
        End Sub


    End Class

End Namespace

File: IdeIntegration/Vs2013Integration/ItemTemplates/SpecFlowStepDefinition_FSharp/SpecFlowStepDefinition1.fs
[<TechTalk.SpecFlow.Binding>]
module $safeitemname$

open TechTalk.SpecFlow

// For additional details on SpecFlow step definitions see http://go.specflow.org/doc-stepdef

let [<Given>] ``I have entered (.*) into the calculator``(number:int) = 
    ScenarioContext.Current.Pending() //TODO: implement arrange (precondition) logic

let [<When>] ``I press add``() = 
    ScenarioContext.Current.Pending() //TODO: implement act (action) logic

let [<Then>] ``the result should be (.*) on the screen``(result:int) = 
    ScenarioContext.Current.Pending() //TODO: implement assert (verification) logic
